systemtest/src/main/java/io/strimzi/systemtest/performance/utils/TopicOperatorPerformanceUtils.java

```diff
@@ -0,0 +1,295 @@
+                    LOGGER.info("Remaining KafkaTopic's to delete from {} to {}: {} !", start, end, numberOfTopicsToDelete);
+                    return numberOfTopicsToDelete == 0;
+                } catch (Exception e) {
+                    LOGGER.error("Error while checking for topic deletion: ", e);
+                    return e.getMessage().contains("Not Found") || e.getMessage().contains("the server doesn't have a resource type");
+                }
+            });
+    }
+
+    /**
+     * Manages the full lifecycle of Kafka topics concurrently using a fixed thread pool.
+     * This method processes creation, modification, and deletion for each topic in separate threads,
+     * ensuring all topics are handled simultaneously across available CPU resources.
+     *
+     * +-----------------------------------------------------------------------------+
+     * |                            processAllTopicsConcurrently                     |
+     * +-----------------------------------------------------------------------------+
+     * | +-------------+       +-------------+             +-------------+           |
+     * | | Topic 1     |       | Topic 2     |     ...     | Topic N     |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | | | Thread  | |       | | Thread  | |             | | Thread  | |           |
+     * | | |         | |       | |         | |             | |         | |           |
+     * | | | Creation| |       | | Creation| |             | | Creation| |           |
+     * | | | Update  | |       | | Update  | |             | | Update  | |           |
+     * | | | Deletion| |       | | Deletion| |             | | Deletion| |           |
+     * | | +---------+ |       | +---------+ |             | +---------+ |           |
+     * | +-------------+       +-------------+             +-------------+           |
+     * +-----------------------------------------------------------------------------+
+     *
+     * @param testStorage           An instance of TestStorage containing configuration and state needed for topic operations.
+     * @param numberOfTopics        The number of Kafka topics to be processed.
+     * @param spareEvents           The number of spare events to be consumed during the process.
+     * @param warmUpTasksToProcess  The number of tasks to warm-up performance and optimize JIT. This number is used just for offsetting.
+     *
+     * @return                      The total time taken to complete all topic lifecycles in milliseconds.
+     */
+    public static long processAllTopicsConcurrently(TestStorage testStorage, int numberOfTopics, int spareEvents, int warmUpTasksToProcess) {
+        final int availableCPUs = Math.max(1, Runtime.getRuntime().availableProcessors());
+        final ExecutorService executor = Executors.newFixedThreadPool(availableCPUs);
```

Comment:
I also did this wrong, but we could actually initialize this once statically at the class level, and set up a shutdown hook to stop it.

Reply:
Hmm...that way it would be better to have some abstraction....handling this as an object rather than a Util class... :) but I would rather leave for the next PR.

Reply:
IMO, it's a small pragmatic improvement, without a huge impact on test results.

Reply:
Updated